pkg/controller/cluster/main-controller.go

```diff
@@ -745,6 +746,10 @@ func (c *Controller) syncHandler(key string) error {
 
 	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration, c.getTransport())
 	if err != nil {
+		if _, err = c.updateTenantStatus(ctx, tenant, StatusTenantCredentialsNotSet, 0); err != nil {
```

Comment:
```suggestion
		if _, err = c.updateTenantStatus(ctx, tenant, StatusTenantCredentialsNotSet, 0); err != nil {
```
This error shadows the error returned by `tenant.NewMinIOAdmin()` leading to the err being `nil` 

Instead you should use a newer variable such as `uerr` or something.